packages/react-slider/src/components/Slider/useSlider.tsx

```diff
@@ -0,0 +1,265 @@
+  const thumbRef = React.useRef<HTMLDivElement>(null);
+  const disposables = React.useRef<(() => void)[]>([]);
+  const id = useId('Slider', props.id);
+
+  /**
+   * Updates the `currentValue` to the new `incomingValue` and clamps it.
+   *
+   * @param ev
+   * @param incomingValue
+   */
+  const updateValue = React.useCallback(
+    (ev, incomingValue: number): void => {
+      if (incomingValue !== min && incomingValue !== max) {
+        ev.stopPropagation();
+        if (ev.type === 'keydown') {
+          ev.preventDefault();
+        }
+      }
+
+      setCurrentValue(clamp(incomingValue, min, max), ev);
+    },
+    [max, min, setCurrentValue],
+  );
+
+  /**
+   * Calculates the `step` position based off of a `Mouse` or `Touch` event.
+   */
+  const calculateSteps = React.useCallback(
+    (ev): number => {
+      const currentBounds = railRef?.current?.getBoundingClientRect();
+      const size = currentBounds?.width || 0;
+      const position = currentBounds?.left || 0;
+
+      const totalSteps = (max - min) / step;
+      const stepLength = size / totalSteps;
+      const thumbPosition = ev.clientX;
+      const distance = thumbPosition - position;
+      return distance / stepLength;
+    },
+    [max, min, step],
+  );
+
+  const onPointerMove = React.useCallback(
+    (ev): void => {
+      if (snapToStep || step !== 1) {
+        updateValue(ev, Math.round((min + step * calculateSteps(ev)) / step) * step);
+      } else {
+        updateValue(ev, min + step * calculateSteps(ev));
+      }
+    },
+    [calculateSteps, min, snapToStep, step, updateValue],
+  );
+
+  const onPointerUp = (): void => {
+    disposables.current.forEach(dispose => dispose());
+    disposables.current = [];
+  };
+
+  const onPointerDown = React.useCallback(
+    (ev): void => {
+      const { currentTarget, pointerId } = ev;
+
+      if (currentTarget.setPointerCapture) {
+        currentTarget.setPointerCapture(pointerId);
+      }
+
+      disposables.current.push(
+        on(currentTarget, 'pointermove', onPointerMove),
+        on(currentTarget, 'pointerup', onPointerUp),
+        () => currentTarget.releasePointerCapture(pointerId),
+      );
+
+      onPointerMove(ev);
+    },
+    [onPointerMove],
+  );
+
+  const onKeyDown = React.useCallback(
+    (ev: React.KeyboardEvent): void => {
+      const key = getCode(ev);
+
+      if (ev.shiftKey) {
+        if (key === ArrowDownKey || key === ArrowLeftKey) {
+          updateValue(ev, currentValue - step * 10);
+          return;
+        } else if (key === ArrowUpKey || key === ArrowRightKey) {
+          updateValue(ev, currentValue + step * 10);
+          return;
+        }
+      } else if (key === ArrowDownKey || key === ArrowLeftKey) {
+        updateValue(ev, currentValue - step);
+        return;
+      } else if (key === ArrowUpKey || key === ArrowRightKey) {
+        updateValue(ev, currentValue + step);
+        return;
+      } else {
+        switch (getCode(ev)) {
```

Comment:
I think you should be able to do:

```suggestion
        switch (key) {
```